os/arch/arm/src/imxrt/imxrt_gpio.c

```diff
@@ -734,10 +734,32 @@ uint8_t imxrt_gpio_irqvector(uint32_t pinset)
 	int port = (pinset & GPIO_PORT_MASK) >> GPIO_PORT_SHIFT;
 	int pin = (pinset & GPIO_PIN_MASK) >> GPIO_PIN_SHIFT;
 
-	if (port == GPIO_PORT1 >> GPIO_PORT_SHIFT) {
+	switch (port) {
+	case GPIO1INDEX:
+		if (pin > 15) {
+			return _IMXRT_GPIO1_16_31_BASE + (pin - 16);
```

Comment:
IMO, It is good to use the switch statement to assign a value to a variable and break. Later return that variable after the switch statement at function exit.

Reply:
I applied your comment. Thanks